lib/trino-aws/src/main/java/io/trino/aws/AwsCredentialsProviderConfig.java

```diff
@@ -33,9 +37,24 @@
     private Optional<String> awsSecretKey = Optional.empty();
     private Optional<String> awsSessionToken = Optional.empty();
     private Optional<String> iamRole = Optional.empty();
+    private String iamRoleSessionName = DEFAULT_ROLE_SESSION_NAME;
     private Optional<String> awsRegionName = Optional.empty();
+    private Optional<String> awsStsEndpoint = Optional.empty();
     private Optional<String> externalId = Optional.empty();
 
+    public Optional<URI> getAwsServiceUri()
+    {
+        return awsServiceUri;
+    }
+
+    @Config("service-uri")
```

Comment:
Adding this here conflates updating TrinoS3Filesystem to use the common module with adding new features that affect other connectors. Keep logically independent changes separate.